src/grid/SaleGridView.php

```diff
@@ -161,12 +161,15 @@ public function columns()
                 'enableSorting' => false,
                 'format' => 'datetime',
             ],
-            'ticket' => [
-                'attribute' => 'ticket',
+            'reason' => [
+                'attribute' => 'reason',
+                'filter' => false,
                 'format' => 'raw',
-                'value' => fn(Sale $sale): string => $sale->ticket ? Html::a($sale->ticket,
-                    ['@ticket/view', 'id' => $sale->ticket],
-                    ['target' => '_blank']) : '',
+                'value' => fn(Sale $sale): string => $sale->reason ?
+                    (is_numeric($sale->reason) ? Html::a($sale->reason,
+                        ['@ticket/view', 'id' => $sale->reason],
+                        ['target' => '_blank']) : (string) $sale->reason
+                    ) : ''
```

Comment:
```suggestion
                'value' => function (Sale $sale): string {
                     if (!is_numeric($sale->reason)) {
                        return Html::encode($sale->reason);
                     }
                     
                     return Html::a($sale->reason,
                        ['@ticket/view', 'id' => $sale->reason],
                        ['target' => '_blank']
                     );
                }
```

XSS injection here. `$sale->reason` should be encoded.

Don't write overcomplicated arrow functions, they are not readable.